contextual_lenses/test_contextual_lenses/test_pooling.py

```diff
@@ -0,0 +1,134 @@
+          'padding_mask' : jnp.array([[[1], [1], [1]], 
+                                      [[1], [1], [0]]]),
+          'mean_pool_rep': jnp.array([[4, 5, 6, 7],
+                                      [2, -3, 4, -5]]),
+          'max_pool_rep': jnp.array([[8, 9, 10, 11],
+                                     [4, -1, 6, -3]])
+      }
+
+test4 = {
+          'x': jnp.array([[[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]],
+                          [[0, -1, 2, -3], [4, -5, 6, -7], [8, -9, 10, -11]]]),
+          'padding_mask' : jnp.array([[[1], [0], [1]], 
+                                      [[0], [1], [0]]]),
+          'mean_pool_rep': jnp.array([[4, 5, 6, 7],
+                                      [4, -5, 6, -7]]),
+          'max_pool_rep': jnp.array([[8, 9, 10, 11],
+                                     [4, -5, 6, -7]])
+      }
+
+test5 = {
+          'x': jnp.array([[[5, 2, -5], [1, -3, 4], [-3, -8, 1]],
+                          [[-2, -7, 4], [9, -4, 5], [-1, 2, -3]]]),
+          'padding_mask' : None,
+          'mean_pool_rep': jnp.array([[1, -3, 0],
+                                      [2, -3, 2]]),
+          'max_pool_rep': jnp.array([[5, 2, 4],
+                                     [9, 2, 5]])
+      }
+
+test6 = {
+          'x': jnp.array([[[5, 2, -5], [1, -3, 4], [-3, -8, 1]],
+                          [[-2, -7, 4], [9, -4, 5], [-1, 2, -3]]]),
+          'padding_mask' : jnp.array([[[1], [1], [1]], 
+                                      [[1], [1], [1]]]),
+          'mean_pool_rep': jnp.array([[1, -3, 0],
+                                      [2, -3, 2]]),
+          'max_pool_rep':  jnp.array([[5, 2, 4],
+                                     [9, 2, 5]])
+      }
+
+test7 = {
+          'x': jnp.array([[[6, 2, -5], [1, -2, 4], [-3, -8, 0]],
+                          [[-2, -7, 4], [10, -4, 5], [-1, 3, -3]]]),
+          'padding_mask' : jnp.array([[[0], [1], [1]], 
+                                      [[1], [0], [0]]]),
+          'mean_pool_rep': jnp.array([[-1, -5, 2],
+                                      [-2, -7, 4]]),
+          'max_pool_rep': jnp.array([[1, -2, 4],
+                                     [-2, -7, 4]])
+      }
+
+test8 = {
+          'x': jnp.array([[[6, 2, -5], [1, -2, 4], [-3, -8, 0]],
+                          [[-2, -7, 4], [10, -4, 5], [-1, 3, -3]]]),
+          'padding_mask' : jnp.array([[[1], [0], [1]], 
+                                      [[1], [1], [0]]]),
+          'mean_pool_rep': jnp.array([[3/2, -3, -5/2],
+                                      [4, -11/2, 9/2]]),
+          'max_pool_rep': jnp.array([[6, 2, 0],
+                                     [10, -4, 5]])
+      }
+
+tests = (test1,
+         test2,
+         test3,
+         test4,
+         test5,
+         test6,
+         test7,
+         test8)
+
+
+class TestPooling(parameterized.TestCase):
+  """Abstract method for testing mean and max pool reduce functions."""
```

Comment:
inaccurate docstring. I'd just say """Tests for mean and max pool reduce functions."""